src/Microsoft.AspNet.Mvc.Core/DefaultControllerActivator.cs

```diff
@@ -3,106 +3,26 @@
 
 using System;
 using System.Collections.Concurrent;
-using System.Collections.Generic;
-using System.Reflection;
-using Microsoft.AspNet.Http;
-using Microsoft.AspNet.Mvc.Core;
-using Microsoft.AspNet.Mvc.ModelBinding;
 using Microsoft.Framework.DependencyInjection;
 
 namespace Microsoft.AspNet.Mvc
 {
     /// <summary>
-    /// Represents the <see cref="IControllerActivator"/> that is registered by default.
+    /// <see cref="IControllerActivator"/> that uses type activation to create controllers.
     /// </summary>
     public class DefaultControllerActivator : IControllerActivator
     {
-        private readonly Func<Type, PropertyActivator<ActionContext>[]> _getPropertiesToActivate;
-        private readonly IDictionary<Type, Func<ActionContext, object>> _valueAccessorLookup;
-        private readonly ConcurrentDictionary<Type, PropertyActivator<ActionContext>[]> _injectActions;
+        private static readonly Func<Type, ObjectFactory> _createControllerFactory =
```

Comment:
@yishaigalatzer it's ugly either way if you want to do it efficiently. In your example you still need a  field to cache the delegate or you're allocating an extra delegate per-call when converting from method-group. IMO the way it's done here is much more readable/simple when the functional body is small.